packages/components/src/components/file/file.component.ts

```diff
@@ -331,10 +382,20 @@ export default class SynFile extends SynergyElement implements SynergyFormContro
     e.preventDefault();
     e.stopPropagation();
 
-    const files = e.dataTransfer?.files;
+    if (!e.dataTransfer) {
+      return;
+    }
+
+    let { files } = e.dataTransfer;
+
+    if (this.webkitdirectory) {
+      // in case of a folder drop, overwrite the files with the folder content
+      const items = e.dataTransfer?.items;
+      files = await this.handleTransferItems(items);
+    }
 
     if (files) {
```

Comment:
You could write this with lower indentation levels by using early returns, e.g. something like this:


```suggestion
this.userIsDragging = false;

    if (!files) {
      return;
    }

    if (!this.multiple && !this.webkitdirectory && files.length > 1) {
      this.emit('syn-error');
      return;
    }
    // Use the transferred file list from the drag drop interface
    const hasTrigger = this.hasSlotController.test('trigger');
    if (!hasTrigger) {
      const disappearAnimation = getAnimation(this.inputChosen, 'file.text.disappear', { dir: this.localize.dir() });
      const appearAnimation = getAnimation(this.inputChosen, 'file.text.appear', { dir: this.localize.dir() });

      if (this.droparea) {
        const dropIconAnimation = getAnimation(this.dropareaIcon, 'file.iconDrop', { dir: this.localize.dir() });
        // eslint-disable-next-line @typescript-eslint/no-floating-promises
        animateTo(this.dropareaIcon, dropIconAnimation.keyframes, dropIconAnimation.options);
      }
      // eslint-disable-next-line max-len
      await animateTo(this.inputChosen, disappearAnimation.keyframes, disappearAnimation.options);
      this.handleFiles(files);
      await animateTo(this.inputChosen, appearAnimation.keyframes, appearAnimation.options);
    } else {
      this.handleFiles(files);
    }

    this.input.dispatchEvent(new Event('change'));
```